playwright/tests/e2e/steps/transactions-overview.steps.ts

```diff
@@ -0,0 +1,221 @@
+    await transactionsTableValueEpoch.click();
+  }
+);
+Then(/^the user should be redirected to the epoch detail page of the selected epoch number$/, async ({ page }) => {
+  const deatailPageTitle = page.getByTestId("detail.page.title");
+  await expect(deatailPageTitle, "Check title on epoch detail").toHaveText("Epoch Details");
+});
+
+Given(/^the user is in the Transactions page for go to address detail page$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+When(
+  /^the user selects the input or output address of one record of the transactions table in the transactions page$/,
+  async ({ page }) => {
+    const transactionsTableValueEpoch = page.getByTestId("transaction.table.value.inputAddress#0").first();
+    addressId = await transactionsTableValueEpoch.getAttribute("href");
+    await transactionsTableValueEpoch.click();
+  }
+);
+Then(/^the user should be redirected to the address details page of the selected address hash$/, async ({ page }) => {
+  await transactionDetailPage(page).checkAddressDetail({ addressId: addressId || "" });
+});
+
+Then(/^the user should be redirected to the address detail page of the selected address hash$/, async ({ page }) => {
+  const deatailPageTitle = page.getByTestId("detail.page.title");
+  await expect(deatailPageTitle, "Check title on address detail").toHaveText("Address Details");
+});
+
+Given(
+  /^the user is in the Transactions page for go to transaction detail from detail button in widget$/,
+  async ({ page }) => {
+    await transactionOverviewPage(page).goToTransactions();
+  }
+);
+Given(
+  /^the user open the info widget of one transaction record in the transactions table of transaction page$/,
+  async ({ page }) => {
+    await transactionOverviewPage(page).openLastestTransactionWidget();
+  }
+);
+When(/^the user selects the view details button in the transcation info widget$/, async ({ page }) => {
+  const transactionDetailFromWidgetDetailViewBtn = page.getByTestId("transaction.detailViewEpoch.viewDetail");
+  transactionHash = await transactionDetailFromWidgetDetailViewBtn.getAttribute("href");
+  await transactionDetailFromWidgetDetailViewBtn.click();
+});
+Then(/^the transaction details page of the selected transaction should be opened$/, async ({ page }) => {
+  await transactionDetailPage(page).checkTransactionsDetail({ transactionHash: transactionHash || "" });
+});
+
+Given(/^the user is in the Transactions page to address detail from detail button in widget$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+Given(/^the user open the info widget of one transaction record in transaction web$/, async ({ page }) => {
+  await transactionOverviewPage(page).openLastestTransactionWidget();
+});
+When(/^the user selects the Input or Output address hash in the transcation info widget$/, async ({ page }) => {
+  const addressDetailProducer = page.getByTestId("transaction.widget.inputAddress");
```

Comment:
let's keep locators interactions inside the pages functions and returns the values that you could need in the next steps if need it.